### PR TITLE
[WIP] Animate the toggle switches for turning on/off plugins.

### PR DESCRIPTION
**Pull request by RubenG123**
Changes the toggle switches in the plugin list to animate to their new state when clicked. Also cleaned up the plugin list item code a bit as there were some redundant calls in a few places.

I removed the event handler that refreshes the plugin list to enable this to work. It doesn't really seem to be nesesarry to refresh the whole thing when one plugin gets changed. But let me know if you know otherwise!

Credit to @psikoi for the animation frame drawings which he gave me a very long time ago

-----------

Marked as WIP as I need to make a fairly large refactor with the implementation

### CODE REVIEW

**Comment by Adam-** [comment]
see also #3116

**Comment by RubenG123** [comment]
Force pushes were to fix formatting to pass CI

File: runelite-client/src/main/java/net/runelite/client/plugins/config/ConfigPlugin.java [hunk 1]
```diff
@@ -103,12 +102,6 @@ protected void shutDown() throws Exception
 		clientToolbar.removeNavigation(navButton);
 	}
 
-	@Subscribe
-	public void onPluginChanged(PluginChanged event)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/config/ConfigPlugin.java]: Why was this removed?

Comment (RubenG123) [reply runelite-client/src/main/java/net/runelite/client/plugins/config/ConfigPlugin.java]: I removed the event handler as it was calling Configpanel::refreshPluginList() whenever a single plugin was either enabled or disabled. (This is also the only time its called outside of the ConfigPanel ctor)

```java
void refreshPluginList()
{
	// update enabled / disabled status of all items
	pluginList.forEach(listItem ->
	{
		final Plugin plugin = listItem.getPlugin();
		if (plugin != null)
		{
			listItem.setPluginEnabled(pluginManager.isPluginEnabled(plugin));
		}
	});

	if (showingPluginList)
	{
		openConfigList();
	}
}
```

As you can see its just rebuilding the whole plugin list. Unless this is some hacky fix for an unknown issue, it seems redundant to me. And rebuilding the plugin list prevents me from performing my animations. 

I guess if you wanted to keep this, I could add a parameter to the event that lets the refreshPluginList() method know to fire an animation for the updated plugin?

**Comment by Adam-** [comment]
@psikoi can you review this?

File: runelite-client/src/main/java/net/runelite/client/plugins/config/ConfigPlugin.java (hunk 1 above)
Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/plugins/config/ConfigPlugin.java]: It is here so the panel will update when plugin is toggled outside of config panel (so, for example GPU plugin, or logging in/out of rl account)

Comment (RubenG123) [reply runelite-client/src/main/java/net/runelite/client/plugins/config/ConfigPlugin.java]: @deathbeam Oh right, thanks! I will definitely need to change a lot of my impl then

Comment (RubenG123) [reply runelite-client/src/main/java/net/runelite/client/plugins/config/ConfigPlugin.java]: @deathbeam I can't seem to find any code that toggles the state of plugins outside of actually clicking the button itself, can you point me in the right direction? I also think I may have found a bug with what you are describing

**Comment by RubenG123** [comment]
On second thought, I don't think this is worth doing so I'll just close this for good. Some people may even prefer it to not even be animated.
